### DISCUSSION TITLE
2 HMS gleichzeitig kaskadiert nutzen?

### DISCUSSION THREAD

**Discussion by rs3hc** [hoylabs/OpenDTU-OnBattery#1146] [General]
Hallo Leute,

Es ist ja möglich mehrere HMS Wechselrichter gleichzeitig zu verwenden.
Korrigiert mich bitte, aber es wird doch am Ende nur der eine geregelt welchen ich in den Settings festlege??

Ist es möglich das leistungsabhängig kaskadiert zu regeln?

Angenommen man hat zwei 300W WR.
0-300W übernimmt WR1 und bei allen darüber wird dann der Zweite geregelt.

**Comment by spcqike** [comment]
Morgen

> Korrigiert mich bitte, aber es wird doch am Ende nur der eine geregelt welchen ich in den Settings festlege??

korrekt.

> Ist es möglich das leistungsabhängig kaskadiert zu regeln?

bisher nicht.

Alternativen: 
- 2 openDTUs
- externe Steuerung, die den 2. Wechselrichter startet und stoppt

letzteres wäre aktuell meine bevorzugte Wahl. der 1. WR wird dauerhaft vom DPL geregelt, der 2. wird bei Bedarf auf 100% zugeschalten. Dadurch würde der DPL den 1. ja runter regeln. Die Regelung kann extern in HA/Node-Red oder dergleichen passieren. Es gibt aber auch irgendwo ein Script, was auf einem Shelly läuft, welches das kann. Ich find es nur auf die Schnelle nicht :)

@Manos1966 weißt du noch, wo das Script war? Im Wiki finde ich es nicht.

**Comment by Manos1966** [comment]
Ich habe es im Q&A "versteckt": https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/FAQ#can-opendtu-onbattery-dynamic-power-limiter-control-more-than-one-inverter

**Note: Member @genmad has made available a Test-script to handle many inverters, using a Shelly PRO 3EM** _(actually, any Shelly with API Gen.2 is suitable to run the script)_: [shellyAPIGen2AndAboveScripts](https://github.com/genmad/shellyAPIGen2AndAboveScripts)

**Comment by schlimmchen** [comment]
Sprechen wir hier von solargespeisten WR oder batteriegespeisten, oder eine Mischung? Und was kann dieses Skript davon handhaben?

**Reply by Manos1966** [reply]
Beides.

Das Skript macht eine ganz simple Sache: 
- Nimmt die Leistung des Stromzaehlers
- teilt sie auf (unter Berucksichtigung verschiedenen Faktoren usw, usw)
-  sendet die berechneten Werte als "Power Meter" an jeweils eine OpenDTU-OnBattery pro Wechselrichter

**Reply by spcqike** [reply]
> an jeweils eine OpenDTU-OnBattery pro Wechselrichter

oh, dann hatte ich das sogar falsch in Erinnerung. Ich dachte, das sendet die Leistungsvorgabe der beiden WR an eine openDTU. 

Dann würde ich wahrscheinlich wirklich auf Node-Red setzen und einfach einen WR per DPL regeln lassen und wenn dieser bspw. 90% belegt ist, den zweiten dazu schalten. und den zweiten wieder abschalten, wenn der 1. nur 20% hat. 

die genauen Grenzen kommen halt auf die Leistungen der WR an, ich habe einen HMS-2000 und einen HM-600. (letzterer ist aktuell nicht in Betrieb, daher auch keine Regelung :D) aber würde ich es machen wollen, würde ich es so machen: 
bei 1.800W den HM-600 dazu schalten, der HMS-2000 geht auf 1.200W runter und hat wieder "Luft". Fällt der HMS-2000 auf 400W, würde der HM-600 wieder deaktiviert und der HMS-2000 "steigt" zrück auf 1.000W. irgendwie so.

**Reply by Manos1966** [reply]
> Ich dachte, das sendet die Leistungsvorgabe der beiden WR an eine openDTU.

Nein, weil wir festgestellt haben, mehrere Wechselrichter = größere Verzögerung des Sendens der Befehle!
Da war eine Geschichte mit den vielen Wiederholungen des Befehls... erinnerst du an die Wasserkocher/Herdplatte Diskussion? 😉

**Reply by rs3hc** [reply]
> > Ich dachte, das sendet die Leistungsvorgabe der beiden WR an eine openDTU.
> 
> Nein, weil wir festgestellt haben, mehrere Wechselrichter = größere Verzögerung des Sendens der Befehle!
> Da war eine Geschichte mit den vielen Wiederholungen des Befehls... erinnerst du an die Wasserkocher/Herdplatte Diskussion? 😉 
> 

Wobei doch sinnvollerweise die Maximale Anzahl an WR doch 3 sein sollte.
Da kannst du schon auf die Phasen aufteilen.

Hier hoch ist denn da der Sendeintervall?
Ich würde ja die Leistung ah nur alle 5-15 Sekunden ändern.🤔

**Reply by genmad** [reply]
> > Ich dachte, das sendet die Leistungsvorgabe der beiden WR an eine openDTU.
> 
> Nein, weil wir festgestellt haben, mehrere Wechselrichter = größere Verzögerung des Sendens der Befehle! Da war eine Geschichte mit den vielen Wiederholungen des Befehls... erinnerst du an die Wasserkocher/Herdplatte Diskussion? 😉

das mit der Verzögerung des sendens der Befehle ist für mich eine neue Information. Wann werden die Befehle verzögert? Wenn man 2 Wrs an einer dtu gleichzeitig steuert, oder auch schon wenn 2 Wrs registriert sind und nur einer angesteuert wird ( von dtu oder einem der die api der dtu nutzt.)?
Daher sollte man kaskadiert regeln, bzw wie es @spcqike vorschlägt den einen Wr nur zwischen Volllast und Standby last schalten.

**Reply by Manos1966** [reply]
@genmad haengt davon ab, wie nah die DTU am Wechselrichter ist.
Da die Wechselrichter der Reihe nach abgefragt werden, bei vielen Retransmits, je mehr Wechselrichter, desto laenger die Zeit: #392 (Nothing received, resend whole request)

**Reply by genmad** [reply]
Ok, Danke

**Comment by rs3hc** [comment]
Das grundsätzliches Problem ist also, dass die dtu aktuell nur einen WR regeln kann, und die Befehle nicht ID-basierrt an mehrere schicken kann.

Merkwürdig.... mehrere Auslesen funktioniert ja auch 🤔

**Reply by spcqike** [reply]
Ich hab es jetzt nicht getestet, aber ich denke, dass du bspw per MQTT unterschiedliche Limits an unterschiedliche inverter schicken kannst. Genauso wie du sie starten und stoppen kannst. 

Der DPL hingegen berücksichtigt bisher immer nur einen inverter. 

Für die originale openDTU gibt es ja auch diverse Python scripte die eine Nulleinspeisung auch mit mehreren invertern machen. 

Das Thema ist in der openDTU onBattery, aufgrund der Batterie und ihrer Absicherung, oder gar dem Betrieb mehrerer wechselrichter an mehreren Batterien oder auch ne Mischung aus Batterie und Solar  only aber komplizierter.

**Reply by Manos1966** [reply]
> Das grundsätzliches Problem ist also, dass die dtu aktuell nur einen WR regeln kann, und die Befehle nicht ID-basiert an mehrere schicken kann.
Merkwürdig.... mehrere Auslesen funktioniert ja auch 

Nein, das grundsätzliche Problem ist, dass es lange dauern **könnte**, Befehle an mehreren Wechselrichtern **erfolgreich** zu senden.
Fuer mehr Details, muss du die Diskussionen lesen... #830

**Reply by Tiese** [reply]
@rs3hc 
Oder man nimmt **_zusätzlich_** das [hier](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/1108)!
